app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -2062,7 +2062,9 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void advance() {
-        next();
+        if (!questionsView.isQuestionList() && canNavigateForward()) {
```

Comment:
@wpride This means that all usages of auto-advance would now only do so if it's not a question list, and if it's not on the last screen. I think it's easy to agree on the first condition being correct. The second is a little harder but also seems correct, since I don't think the act of answering a question should ever cause a form to submit.